Tests/src/PublicAPITests/CancellationTokenTests.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Awaitable.h"
+#include "CSP/CSPFoundation.h"
+#include "CSP/Common/CancellationToken.h"
+#include "TestHelpers.h"
+
+#include "gtest/gtest.h"
+
+namespace
+{
+
+#if RUN_ALL_UNIT_TESTS || RUN_CANCELLATION_TOKEN_TESTS || RUN_CANCELLATIONTOKEN_CONSTRUCTION_TEST
+CSP_PUBLIC_TEST(CSPEngine, CancellationTokenTests, ConstructionAndDestructionTest)
+{
+	csp::common::CancellationToken CancellationToken;
+	// No need to assert anything, just ensuring that construction and destruction don't crash
+}
+#endif
+
+#if RUN_ALL_UNIT_TESTS || RUN_CANCELLATION_TOKEN_TESTS || RUN_CANCELLATIONTOKEN_CANCEL_TEST
+CSP_PUBLIC_TEST(CSPEngine, CancellationTokenTests, CancelStateTest)
+{
+	csp::common::CancellationToken CancellationToken;
+	EXPECT_FALSE(CancellationToken.Cancelled());
+
+	CancellationToken.Cancel();
+	EXPECT_TRUE(CancellationToken.Cancelled());
+
+	CancellationToken.Cancel(); // Test that multiple cancellations don't affect the state
+	EXPECT_TRUE(CancellationToken.Cancelled());
+}
+#endif
+
+#if RUN_ALL_UNIT_TESTS || RUN_CANCELLATION_TOKEN_TESTS || RUN_CANCELLATIONTOKEN_COPYMOVE_TEST
+CSP_PUBLIC_TEST(CSPEngine, CancellationTokenTests, CopyMoveTest)
+{
+	// Ensure copy and move operations are deleted
+	ASSERT_FALSE(std::is_copy_constructible<csp::common::CancellationToken>::value);
```

Comment:
Prefer the `std::is_*_v<T>` forms over `std::is_*<T>::value`

Reply:
No issue this, but interested in the rationale, since the former is a typedef of the latter. Is there a portability reason?

Reply:
Also addressed in latest.

Reply:
The C++ committee suggests using them since they were introduced, as it's less to type and also more immediately obvious what it is you're doing. Plus, you only ever use these structs when wanting to get the value from them, so it makes sense to use the less verbose form.

Reply:
Got it, thanks! 👍